Libraries/LibWeb/DOM/Node.cpp

```diff
@@ -2275,6 +2277,12 @@ ErrorOr<String> Node::name_or_description(NameOrDescription target, Document con
                 total_accumulated_text.append(result);
             }
             // iii. Return the accumulated text.
+            // AD-HOC: This substep in the spec doesn’t seem to explicitly require the following check for an aria-label
+            // value; but the “button's hidden referenced name (visibility:hidden) with hidden aria-labelledby traversal
+            // falls back to aria-label” subtest at https://wpt.fyi/results/accname/name/comp_labelledby.html won’t pass
+            // unless we do this check.
+            if (total_accumulated_text.string_view().is_whitespace() && target == NameOrDescription::Name && element->aria_label().has_value() && !element->aria_label()->is_empty() && !element->aria_label()->bytes_as_string_view().is_whitespace())
```

Comment:
Hmm, not really a comment on your work, but we should probably have `is_whitespace()` (or `is_ascii_whitespace()`?) support for `String` directly.

Reply:
I am working on a more general solution for that in #2442 that also adds a `is_ascii_whitespace`, but you can definitely add that, it should remain backwards/forwards compatible with my PR.